### ISSUE TITLE
Typescript 4.7 instantiation expressions break with "Unexpected token"

### ISSUE THREAD

**Issue by chrisui** [prettier/prettier#13225]
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEEDWBlGAJ0wF5scAeDQgSygHMA+AHSkig0zkMKgFcAtqS48IhCvwEAjboxAAaEBAAOMGumSgAhqIDuABR0I0yEFoA2urQE8TiqYS1gccGHi0C4AGTpxkAMws0OHtHZ1c8ZSc6emQiPhCQOGk4ABNUtK8tBj4tejgAMTEBLRg1BlMtPhgIBRAACxgBcwB1epp4NCiwODxjDpoANw7rUzA0OxA6YMIYfUd6EoCgxIArNAAPPBjzOABFPgh4ZfNgxSjCGdMpLRlzOuVaWBaaVJh65AAOAAZzwghgi1HMpTI84DNBn5FABHQ7weYqEwoLRoAC0UDgaTSdUIcFhNFx8zySyQgVOiWCAhocUICUUaB2+zhflJK0UMFuLzeHyQACZ2Y4aOYYgBhCACElJNAAVjqfGCABVbkiyWcQIMEgBJKAZWB4MC0VQAQR1BGsuxOwQAvlagA) (& [equivilent ts playground](https://www.typescriptlang.org/play?jsx=0&ts=4.7.4#code/MYewdgzgLgBCDWMC8MA8AVAfACgG4C4Z0BKZTGXAKFElgFMAnBkB5NLPQksiy68aHHgBlKKxQJU0BgEswAc0z9aMRgzABXALZs1LVJq0AjRpiA))

```sh
--parser babel
```

**Input:**

```jsx
const okStr = ok<string>
const errnum = error<number>
```

**Output:**

```jsx
SyntaxError: Unexpected token (2:1)
  1 | const okStr = ok<string>
> 2 | const errnum = error<number>
    | ^
```

**Expected behavior:**

No error, formats happily without semi-colons.

https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#instantiation-expressions

**Comment by chrisui** [comment]
Note this particular code works when forcing the parser to "typescript" however I'm not sure how to do that on a project configuration level with many parsers required. Seems the default inferred parser is incorrect in this case as I see this error locally with no "parser" option configured.

**Comment by chrisui** [comment]
Not also seeing errors in usage elsewhere:

```
[error] shared/lib/src/result.ts: SyntaxError: Expression expected. (74:36)
[error]   72 |   // as casting is used here purely for refinement as type
[error]   73 |   // instantiations cannot yet be used in a arg position
[error] > 74 |   Ok = ok as typeof ok<OkValueType>,
[error]      |                                    ^
[error]   75 |   Error = error as typeof error<ErrorValueType>
[error]   76 | ) => {
[error]   77 |   return variant({ Ok, Error })
```

**Comment by thorn0** [comment]
[Please include a repro](https://gist.github.com/Rich-Harris/88c5fc2ac6dc941b22e7996af05d70ff).

**Comment by chrisui** [comment]
> [Please include a repro](https://gist.github.com/Rich-Harris/88c5fc2ac6dc941b22e7996af05d70ff).

What?

I'm confused. 

The issue template, and my issue, contains a link to several playgrounds and several code snippets. How is this not a repro?

**Comment by thorn0** [comment]
TypeScript isn't formatted if you don't choose the TypeScript parser. This is expected behavior. What exactly is the supposed bug? How can I reproduce it?

**Comment by chrisui** [comment]
Great, thank you. Much more helpful than a passive-aggressive link that gives zero guidance on why what I provided didn't quite cut it because that required context I clearly did not have.

[Here is a repro.](https://github.com/chrisui/prettier-repros) which was triggered only _after_ I added a `.prettierrc` which includes two plugins. Turns out the issue is going to be something in [prettier-plugin-jsdoc](https://github.com/hosseinmd/prettier-plugin-jsdoc).

Will close this issue.
